sourcemap/src/lib.rs

```diff
@@ -114,46 +115,60 @@ impl SourceMapView {
     /// If the sourcemap is an index it is being flattened.  If flattening
     /// is not possible then an error is raised.
     pub fn from_json_slice(buffer: &[u8]) -> Result<Self, ParseSourceMapError> {
+        use sourcemap::DecodedMap::*;
         Ok(SourceMapView {
-            sm: match sourcemap::decode_slice(buffer)? {
-                sourcemap::DecodedMap::Regular(sm) => sm,
-                sourcemap::DecodedMap::Index(smi) => smi.flatten()?,
+            dm: match sourcemap::decode_slice(buffer)? {
+                Regular(sm) => Regular(sm),
+                Index(smi) => Regular(smi.flatten()?),
+                Hermes(smh) => Hermes(smh),
+                _ => panic!("invalid sourcemap type"),
             },
         })
     }
 
+    fn as_sm(&self) -> &sourcemap::SourceMap {
+        use sourcemap::DecodedMap::*;
+        match &self.dm {
+            Regular(sm) => sm,
+            Hermes(smh) => smh,
+            _ => unreachable!(),
```

Comment:
Rather than having `unreachable!()` here, can we introduce our own enum instead? Or, alternatively, store the raw sourcemap plus a plain enum for `lookup_token_with_function_name`.